packages/merge-on-green/test/merge-on-green.ts

```diff
@@ -324,6 +324,52 @@ describe('merge-on-green-', () => {
       scopes.forEach(s => s.done());
     });
 
+    it.only('rejects status checks that do not match the required check', async () => {
+      const scopes = [
+        getPR(true, 'clean', 'open'),
+        getBranchProtection(["this is what we're looking for"]),
+        getReviewsCompleted([{user: {login: 'octocat'}, state: 'APPROVED'}]),
+        getLatestCommit([{sha: '6dcb09b5b57875f334f61aebed695e2e4193db5e'}]),
+        getMogLabel([{name: 'automerge'}]),
+        getStatusi('6dcb09b5b57875f334f61aebed695e2e4193db5e', [
+          {state: 'success', context: "this is what we're looking fo"},
```

Comment:
```suggestion
          {state: 'success', context: "this is what we're looking for"},
```

Or does this intentionally not match? Might want to change to "not what we're looking for" if that's the case.

Reply:
So, this is an interesting test. It intentionally does not match, but I wanted to make the string as close to the test it is checking as possible (minus one character) to check the directionality of the regex (i.e., only if the status check matches the first part of the required status check will it pass, otherwise it will fail). Of course, it's regex, so it could be an entirely different string and it won't pass either. Changing the string to "not what we're looking for" would make more sense, but then in that case it would be like the regular status check test that fails if the status check that runs does not match the required status check. I feel like this is a bit of a ramble, but hopefully it makes sense. With this explanation in mind, what would you recommend? We could take out this test entirely, given that it's pretty much covered by other tests that check to see if the required status checks ran and passed.

Reply:
I added in a comment explaining the situation in the code, but I'd also love to hear any other suggestions you might have!